test/config_test.rb

```diff
@@ -0,0 +1,196 @@
+require_relative "test_helper"
+
+class ConfigTest < Minitest::Test
+  include TestHelper
+
+  def test_content_without_sider_yml
+    mktmpdir do |path|
+      assert_equal({}, Runners::Config.new(path).content)
+    end
+  end
+
+  def test_content_with_empty_sider_yml
+    mktmpdir do |path|
+      (path / "sider.yml").write("---")
+      assert_equal({}, Runners::Config.new(path).content)
+    end
+
+    mktmpdir do |path|
+      (path / "sider.yml").write("")
+      assert_equal({}, Runners::Config.new(path).content)
+    end
+  end
+
+  def test_content_with_linter_section
+    mktmpdir do |path|
+      (path / "sider.yml").write(<<~YAML)
+        ---
+        linter:
+          eslint:
+            config: abc
+            options:
+              npm_install: true
+      YAML
+      assert_equal(
+        { linter: {
+          brakeman: nil,
+          checkstyle: nil,
+          code_sniffer: nil,
+          coffeelint: nil,
+          cppcheck: nil,
+          cpplint: nil,
+          eslint: {
+            root_dir: nil,
+            npm_install: nil,
+            dir: nil,
+            ext: nil,
+            config: "abc",
+            "ignore-path": nil,
+            "ignore-pattern": nil,
+            "no-ignore": nil,
+            global: nil,
+            quiet: nil,
+            options: { npm_install: true,
+                       dir: nil,
+                       ext: nil,
+                       config: nil,
+                       "ignore-path": nil,
+                       "no-ignore": nil,
+                       "ignore-pattern": nil,
+                       global: nil,
+                       quiet: nil,
+            },
+          },
+          flake8: nil,
+          fxcop: nil,
+          go_vet: nil,
+          golangci_lint: nil,
+          golint: nil,
+          gometalinter: nil,
+          goodcheck: nil,
+          hadolint: nil,
+          rubocop: nil,
+          haml_lint: nil,
+          javasee: nil,
+          jshint: nil,
+          ktlint: nil,
+          misspell: nil,
+          phinder: nil,
+          phpmd: nil,
+          pmd_java: nil,
+          querly: nil,
+          rails_best_practices: nil,
+          reek: nil,
+          scss_lint: nil,
+          shellcheck: nil,
+          stylelint: nil,
+          swiftlint: nil,
+          tslint: nil,
+          tyscan: nil },
+          ignore: nil,
+          branches: nil,
+        },
+        Runners::Config.new(path).content,
+      )
+    end
+  end
+
+  def test_content_with_unknown_linter
+    mktmpdir do |path|
+      (path / "sider.yml").write(<<~YAML)
+        ---
+        linter:
+          unknown_linter:
+            config: abc
+      YAML
+      assert_raises Runners::Config::InvalidConfiguration do
+        Runners::Config.new(path)
+      end
```

Comment:
[suggestion] What about testing also an error message? Because the message depends on StrongJSON methods which may be changed in the future.

```suggestion
      exn = assert_raises Runners::Config::InvalidConfiguration do
        Runners::Config.new(path)
      end
      assert_equal "The attribute `$.linter.unknown_linter` of `sider.yml` cannot exist.", exn.message
```

Reply:
OK, looks good.